src/features/contacts/contacts-table.tsx

```diff
@@ -1,60 +1,185 @@
-import { Button } from "@mui/material";
-import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import {
+  DataGrid,
+  getGridStringOperators,
+  GridColDef,
+  GridFilterModel,
+  GridSortModel,
+} from "@mui/x-data-grid";
+import EditIcon from "@mui/icons-material/Edit";
+import ArrowForwardIcon from "@mui/icons-material/ArrowForward";
 import { ContactDetailsDto } from "lib/network/swagger-client";
-import { ContactsTableContainer } from "./index.styled";
+import {
+  ActionButtonContainer,
+  ContactsTableContainer,
+  EditIconContainer,
+  ForwardIconContainer,
+} from "./index.styled";
+import { CoreModule, getEditModuleRoute } from "lib/router";
+import { useMemo, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Avatar, ListItem, ListItemAvatar, ListItemText } from "@mui/material";
 
 type ContactsTableProps = {
   contacts?: ContactDetailsDto[];
+  pageSize?: number;
+  totalRowCount: number;
```

Comment:
Better to make basic table control, which can be inhereted and used for different entity

Reply:
Fixed